clients/drcachesim/tracer/func_trace.cpp

```diff
@@ -125,24 +125,53 @@ func_post_hook(void *wrapcxt, void *user_data)
     append_entry(drcontext, TRACE_MARKER_TYPE_FUNC_RETVAL, retval);
 }
 
+static app_pc
+get_pc_by_symbol(const module_data_t *mod, const char *symbol)
+{
+    // Try to find the symbol in the dynamic symbol table.
+    app_pc pc = (app_pc)dr_get_proc_address(mod->handle, symbol);
+    if (pc != NULL) {
+        dr_fprintf(STDERR, "dr_get_proc_address found symbol %s at pc=%p\n", symbol, pc);
+        return pc;
+    } else {
+        // If failed to find the symbol in the dynamic symbol table, then we try to find
+        // it in the module loaded by reading the module file in mod->full_path.
+        // NOTE: mod->full_path could be invalid in the case where the original
+        // module file is remapped and deleted (e.g. hugepage_text).
+        // FIXME: find a way to find the PC of the symbol even if the original module file
+        // is deleted.
+        size_t offset;
+        drsym_error_t err =
+            drsym_lookup_symbol(mod->full_path, symbol, &offset, DRSYM_DEMANGLE);
+        if (err == DRSYM_SUCCESS) {
+            pc = mod->start + offset;
+            dr_fprintf(STDERR, "drsym_lookup_symbol found symbol %s at pc=%p\n", symbol,
+                       pc);
+            return pc;
+        } else {
+            dr_fprintf(STDERR, "Failed to find symbol %s, drsym_error_t=%d\n", symbol,
+                       err);
+            return NULL;
+        }
+    }
+}
+
 static void
 instru_funcs_module_load(void *drcontext, const module_data_t *mod, bool loaded)
 {
     if (drcontext == NULL || mod == NULL)
         return;
-    dr_log(NULL, DR_LOG_ALL, 1, "instru_funcs_module_load start=%p, mod->full_path=%s\n",
-           mod->start, mod->full_path);
 
+    dr_fprintf(STDERR, "instru_funcs_module_load, mod->full_path=%s\n", mod->full_path);
```

Comment:
full_path can be nullptr: safest to check for that and print "" if so

Reply:
Done.